packages/astro/test/sessions.test.js

```diff
@@ -1,13 +1,17 @@
 import assert from 'node:assert/strict';
 import { before, describe, it } from 'node:test';
+import * as devalue from 'devalue';
 import testAdapter from './test-adapter.js';
 import { loadFixture } from './test-utils.js';
+import fs from 'node:fs';
 
 describe('Astro.session', () => {
 	/** @type {import('./test-utils').Fixture} */
 	let fixture;
 
 	before(async () => {
+		// seems to be a race condition in unstorage when the session directory doesn't exist
+		await fs.promises.mkdir('.astro/session', { recursive: true });
```

Comment:
We should probably be creating this in production too in that case. It could go here: 

https://github.com/kaytwo/astro/blob/6e7afe876fa8ee07457dd60e145de70aa3f9f5c3/packages/astro/src/core/session.ts#L396